KECA-41: add stream command to control streaming call media to websocket

icehess opened the pull request:
Adding dialplan command for start/stop/pause/resume kz_audio_fork and streaming call's media to a WebSocket.

The stream will the media as raw audio with PCMU (mu-law) encoding. Sample rate and number of channels are configurable.

The stream can be stopped, pause or resume. By default if no Stream-ID is given, these commands will act on all attached audio forks.

icehess committed:
add stream command to control streaming call media to websocket

Adding dialplan command for start/stop/pause/resume kz_audio_fork
and streaming call's media to a WebSocket.

The stream will the media as raw audio with PCMU (mu-law) encoding.
Sample rate and number of channels are configurable.

The stream can be stopped, pause or resume. By default if no
Stream-ID is given, these commands will act on all attached
audio forks.

icehess commented:
PRs:
   - master:
       - https://github.com/2600hz/kazoo-core/pull/980
       - https://github.com/2600hz/kazoo-core/pull/1023
       - https://github.com/2600hz/kazoo-ecallmgr/pull/265
       - https://github.com/2600hz/kazoo-ecallmgr/pull/269
   - 5.4:
       - https://github.com/2600hz/kazoo-core/pull/1036
       - https://github.com/2600hz/kazoo-ecallmgr/pull/270

dhuynh-Kazoo committed:
Adding new Stream headers to kapi.dialplan.command.json

icehess committed:
make fmt

lazedo commented on kazoo_call/src/kapps_call_command.erl:
```diff
@@ -1909,6 +1915,86 @@ b_record_call(MediaName, Action, TimeLimit, Terminators, Call) ->
+    stream(Action, [], Call).
+
+-spec stream(kz_term:ne_binary(), kz_term:proplist(), kapps_call:call()) -> 'ok'.
+stream(Action, Options, Call) ->
```
change to `stream_command` to follow other commands

lazedo reviewed (changes requested):
add `block` for start_stream like other commands. `b_start_stream` that sends and waits for return

I would also like to see stream_`action` instead of `action`_stream, `stream_start` instead of `stop_stream`

icehess committed:
adding blocking functions and rename funs to stream_action

icehess committed:
do not execute now, use correct app name

lazedo commented on kazoo_call/src/kapps_call_command.erl:
```diff
@@ -1909,6 +1918,150 @@ b_record_call(MediaName, Action, TimeLimit, Terminators, Call) ->
+stream_command(Action, Options, Call) ->
+    Command = [{<<"Application-Name">>, <<"stream">>}
+              ,{<<"Action">>, Action}
+              ,{<<"Insert-At">>, <<"now">>}
```
commands that expect a return should not set `{<<"Insert-At">>, <<"now">>}`

icehess committed:
please dialyzer

icehess committed:
split audio type to tracks and mix instead of channel-as-stereo

Audui Tracks refers to:
- single: only recrod the audio that freeswitch is receiving from
  call's channel (the audio of person that is talking into their phone)
- mixed: record both the audio that freeswitch is receiving also the
  the audio that freeswitch is also writing into channel (either the
  audio from other legs or fs is producing/playing on the channel)

Audio Mix indicates how track(s) should be mixed in recroded media:
- mono: all track(s) will be records in single audio channel
- stereo: reading audio will recorded on left channel and writing audio
  will be recorded on right audio channel.
